plugins/ua_securitypolicy_basic128rsa15.c

```diff
@@ -0,0 +1,1299 @@
+        return UA_STATUSCODE_BADCERTIFICATEUSENOTALLOWED;
+
+    return UA_STATUSCODE_GOOD;
+}
+
+static void
+channelContext_deleteContext_sp_basic128rsa15(void *contextData) {
+
+    // Delete the member variables that eventually were allocated in the init method
+    UA_SP_basic128rsa15_ChannelContextData *const data =
+        (UA_SP_basic128rsa15_ChannelContextData *) contextData;
+
+    UA_ByteString_deleteMembers(&data->localSymSigningKey);
+    UA_ByteString_deleteMembers(&data->localSymEncryptingKey);
+    UA_ByteString_deleteMembers(&data->localSymIv);
+
+    UA_ByteString_deleteMembers(&data->remoteSymSigningKey);
+    UA_ByteString_deleteMembers(&data->remoteSymEncryptingKey);
+    UA_ByteString_deleteMembers(&data->remoteSymIv);
+
+    mbedtls_x509_crt_free(&data->remoteCertificate);
+
+    UA_free(data);
+}
+
+static UA_StatusCode
+channelContext_newContext_sp_basic128rsa15(const UA_SecurityPolicy *securityPolicy,
+                                           const UA_ByteString *remoteCertificate,
+                                           void **pp_contextData) {
+
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+
+    if(securityPolicy == NULL ||
+       remoteCertificate == NULL || pp_contextData == NULL)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    *pp_contextData = UA_malloc(sizeof(UA_SP_basic128rsa15_ChannelContextData));
+    if(*pp_contextData == NULL)
+        return UA_STATUSCODE_BADOUTOFMEMORY;
+
+    UA_SP_basic128rsa15_ChannelContextData *const contextData =
+        (UA_SP_basic128rsa15_ChannelContextData *) *pp_contextData;
+
+    contextData->policyContext = (UA_SP_basic128rsa15_PolicyContextData *) securityPolicy->policyContext;
+
+    UA_ByteString_init(&contextData->localSymSigningKey);
+    UA_ByteString_init(&contextData->localSymEncryptingKey);
+    UA_ByteString_init(&contextData->localSymIv);
+
+    UA_ByteString_init(&contextData->remoteSymSigningKey);
+    UA_ByteString_init(&contextData->remoteSymEncryptingKey);
+    UA_ByteString_init(&contextData->remoteSymIv);
+
+    mbedtls_x509_crt_init(&contextData->remoteCertificate);
+
+    // TODO: this can be optimized so that we dont allocate memory before parsing the certificate
+    retval |= parseRemoteCertificate_sp_basic128rsa15(*pp_contextData,
+                                                      remoteCertificate);
+    if(retval != UA_STATUSCODE_GOOD) {
+        channelContext_deleteContext_sp_basic128rsa15(*pp_contextData);
+        *pp_contextData = NULL;
+        return retval;
+    }
+
+    return retval;
+}
+
+static UA_StatusCode
+channelContext_setLocalSymEncryptingKey_sp_basic128rsa15(void *contextData,
+                                                         const UA_ByteString *key) {
+    if(key == NULL || contextData == NULL) {
+        fprintf(stderr,
+                "Error while calling channelContext_setLocalEncryptingKey_sp_basic128rsa15."
+                    "Null pointer passed.");

```

Comment:
Use logging?

Reply:
Yeah. I forgot to change this. In an older version of the API the logger wasn't available here.